docs/_docs/2_4_configurations.md

```diff
@@ -83,8 +83,12 @@ Actual value ontained by invoking .toString()</span>
 
 #### KEYGENERATOR_CLASS_OPT_KEY {#KEYGENERATOR_CLASS_OPT_KEY}
   Property: `hoodie.datasource.write.keygenerator.class`, Default: `org.apache.hudi.keygen.SimpleKeyGenerator` <br/>
-  <span style="color:grey">Key generator class, that implements will extract the key out of incoming `Row` object</span>
-  
+  <span style="color:grey">Key generator class, that implements will extract the key out of incoming `Row` object. This config option is used for user-defined KeyGenerator</span>
```

Comment:
Can you also add a line there, that this has higher precedence over keygen type.

Reply:
also, there is no more default for hoodie.datasource.write.keygenerator.class if I understand correctly. can we fix that as well.

Reply:
> also, there is no more default for hoodie.datasource.write.keygenerator.class if I understand correctly. can we fix that as well.

done